### ISSUE TITLE
Button groups NEXT and PREV requires double pressing

### ISSUE THREAD

**Issue by MichaelVLV** [lvgl/lvgl#810]
Hi.

I have created screen with 4 buttons and want to navigate between them with hardware buttons.
After implemented this task on MCU, I noticed when I'm switching from neighboring buttons it requires double pressing. I thought that this is some problem with button. But after adding debounce and totaly excluded possible hardware problems, I have recreated part of naviation on PC simulator. 
And problem repeated.

Steps to reproduce:

1. Run program
2. Press NEXT button (switch from BUTTON_1 to BUTTON_2 - ok)
3. Press NEXT button again. And now (2 to 3) it requires double pressing to switch.

![image](https://user-images.githubusercontent.com/31571833/52432824-2c0e8480-2b14-11e9-8f82-0c0aafdb8eb3.png)

I used this project: 
[https://github.com/littlevgl/pc_simulator_sdl_visual_studio](https://github.com/littlevgl/pc_simulator_sdl_visual_studio)

Whole code (can be pasted in PC simulator):

```C
/**
* @file main
*
*/

/*********************
*      INCLUDES
*********************/
#include <stdlib.h>
#include <stdio.h>
#include <Windows.h>
#include <SDL.h>
#include "lvgl/lvgl.h"
#include "lv_drivers/display/monitor.h"
#include "lv_drivers/indev/mouse.h"
#include "lv_drivers/indev/keyboard.h"
#include "lv_examples/lv_apps/demo/demo.h"
#include "lv_examples/lv_apps/benchmark/benchmark.h"
#include "lv_examples/lv_tests/lv_test_theme/lv_test_theme_1.h"
#include "lv_examples/lv_tutorial/10_keyboard/lv_tutorial_keyboard.h"

/*********************
*      DEFINES
*********************/

/**********************
*      TYPEDEFS
**********************/

/**********************
*  STATIC PROTOTYPES
**********************/
static void hal_init(void);
static int tick_thread(void *data);
static void kaypad_create(void);
static lv_res_t keypad_btn_release(lv_obj_t * btn);
static lv_res_t keypad_btn_press(lv_obj_t * btn);

/**********************
*  STATIC VARIABLES
**********************/
static lv_indev_t * kb_indev;
static lv_indev_t * emulated_kp_indev;  /*The input device of the emulated keypad*/
static uint32_t last_key = 0;
static lv_indev_state_t last_state = LV_INDEV_STATE_REL;

typedef enum {
	START_STATE = 0,

	MAIN_MENU,

	SOME_OTHER_MENU_1,
	SOME_OTHER_MENU_2,

	STATE_MAX,
} menu_scr_t;

lv_group_t * custom_groups[STATE_MAX];

lv_res_t btn_action(lv_obj_t * btn)
{
	uint8_t id = lv_obj_get_free_num(btn);
	printf("%s: btn(%d)\n", __func__, id);
	return LV_RES_OK;
}

static void UI_delete_all_groups(void)
{
	for (size_t i = 0; i < STATE_MAX; i++)
	{
		if (custom_groups[i] != NULL)
		{
			//printf("%s: avo_group[%d]\n", __func__, i);
			lv_group_del(custom_groups[i]);
		}
	}
}

static void UI_prepare_input_group(menu_scr_t group_idx)
{
	UI_delete_all_groups();
	custom_groups[group_idx] = lv_group_create();
	lv_indev_set_group(emulated_kp_indev, custom_groups[group_idx]);
}

static void scr_main_menu(menu_scr_t group_idx)
{
	printf("main_menu\n");

	lv_obj_t * scr = NULL;
	scr = lv_obj_create(NULL, NULL);
	lv_scr_load(scr);

	// ---------- BUTTON 1
	lv_obj_t * btn1 = lv_btn_create(lv_scr_act(), NULL); /*Create a button on the currently loaded screen*/
	lv_obj_set_size(btn1, 144, 24);
	lv_obj_align(btn1, NULL, LV_ALIGN_IN_TOP_MID, 0, 0);
	lv_obj_set_free_num(btn1, 1);
	lv_obj_t * label = lv_label_create(btn1, NULL);
	lv_label_set_text(label, "BUTTON_1");
	lv_btn_set_action(btn1, LV_BTN_ACTION_CLICK, btn_action);
	lv_group_add_obj(custom_groups[group_idx], btn1);

	// ---------- BUTTON 2
	lv_obj_t * btn2 = lv_btn_create(lv_scr_act(), btn1);
	lv_obj_align(btn2, NULL, LV_ALIGN_IN_TOP_MID, 0, 30);
	label = lv_label_create(btn2, NULL);
	lv_obj_set_free_num(btn2, 2);
	lv_label_set_text(label, "BUTTON_2");
	lv_btn_set_action(btn2, LV_BTN_ACTION_CLICK, btn_action);
	lv_group_add_obj(custom_groups[group_idx], btn2);

	// ---------- BUTTON 3
	lv_obj_t * btn3 = lv_btn_create(lv_scr_act(), btn1);
	lv_obj_align(btn3, NULL, LV_ALIGN_IN_TOP_MID, 0, 60);
	label = lv_label_create(btn3, NULL);
	lv_obj_set_free_num(btn3, 3);
	lv_label_set_text(label, "BUTTON_3");
	lv_btn_set_action(btn3, LV_BTN_ACTION_CLICK, btn_action);
	lv_group_add_obj(custom_groups[group_idx], btn3);

	// ---------- BUTTON 4
	lv_obj_t * btn4 = lv_btn_create(lv_scr_act(), btn1);
	lv_obj_align(btn4, NULL, LV_ALIGN_IN_TOP_MID, 0, 90);
	label = lv_label_create(btn4, NULL);
	lv_obj_set_free_num(btn4, 4);
	lv_label_set_text(label, "BUTTON_4");
	lv_btn_set_action(btn4, LV_BTN_ACTION_CLICK, btn_action);
	lv_group_add_obj(custom_groups[group_idx], btn4);
}

// Button handler
static bool buttons_read(lv_indev_data_t * data)
{
	data->key = last_key;
	data->state = last_state;

	//if (last_key != 0 || last_state != 0)
	{
		printf("last_key: %d  | last_state: %d\r", last_key, last_state);
	}

	return false;
}

static void buttons_drv_reg(void)
{
	lv_indev_drv_t kp_drv;
	lv_indev_drv_init(&kp_drv);
	kp_drv.type = LV_INDEV_TYPE_KEYPAD; //LV_INDEV_TYPE_KEYPAD;  //LV_INDEV_TYPE_BUTTON;
	kp_drv.read = buttons_read;
	emulated_kp_indev = lv_indev_drv_register(&kp_drv);
}

int main(int argc, char** argv)
{
	/*Initialize LittlevGL*/
	lv_init();

	/*Initialize the HAL for LittlevGL*/
	hal_init();
	buttons_drv_reg();

	UI_prepare_input_group(MAIN_MENU);
	scr_main_menu(MAIN_MENU);

	kaypad_create();

	/*Load a demo*/
	//demo_create();

	/*Try the benchmark to see how fast is your GUI*/
	//benchmark_create();

	/*Check the themes too*/
	//lv_test_theme_1(lv_theme_night_init(210, NULL));

	/*A keyboard control example*/
	//lv_tutorial_keyboard(kb_indev);

	while (1) {
		/* Periodically call the lv_task handler.
		* It could be done in a timer interrupt or an OS task too.*/
		lv_task_handler();
		Sleep(10);       /*Just to let the system breathe */
	}

	return 0;
}

/**********************
*   STATIC FUNCTIONS
**********************/

static void kaypad_create(void)
{
	/*Next button*/
	lv_obj_t * btn_next = lv_btn_create(lv_scr_act(), NULL);
	lv_btn_set_action(btn_next, LV_BTN_ACTION_PR, keypad_btn_press);
	lv_btn_set_action(btn_next, LV_BTN_ACTION_CLICK, keypad_btn_release);
	lv_btn_set_action(btn_next, LV_BTN_ACTION_LONG_PR, keypad_btn_release);
	lv_obj_set_size(btn_next, 50, 20);
	lv_obj_set_free_num(btn_next, LV_GROUP_KEY_NEXT);
	lv_obj_t * l = lv_label_create(btn_next, NULL);
	lv_label_set_text(l, "Next");
	lv_obj_align(btn_next, NULL, LV_ALIGN_IN_BOTTOM_LEFT, 10 , 0 );

	/*Increment button*/
	lv_obj_t * btn_inc = lv_btn_create(lv_scr_act(), btn_next);
	lv_obj_set_free_num(btn_inc, LV_GROUP_KEY_PREV);
	l = lv_label_create(btn_inc, NULL);
	lv_label_set_text(l, "Prev");
	lv_obj_align(btn_inc, NULL , LV_ALIGN_IN_BOTTOM_LEFT, 70 , 0);

	/*Decrement button*/
	lv_obj_t * btn_dec = lv_btn_create(lv_scr_act(), btn_next);
	lv_obj_set_free_num(btn_dec, LV_GROUP_KEY_ESC);
	l = lv_label_create(btn_dec, NULL);
	lv_label_set_text(l, "ESC");
	lv_obj_align(btn_dec, NULL, LV_ALIGN_IN_BOTTOM_RIGHT, -10 , 0);

	/*Enter button*/
	lv_obj_t * btn_enter = lv_btn_create(lv_scr_act(), btn_next);
	lv_obj_set_free_num(btn_enter, LV_GROUP_KEY_ENTER);
	l = lv_label_create(btn_enter, NULL);
	lv_label_set_text(l, "Enter");
	lv_obj_align(btn_enter, NULL, LV_ALIGN_IN_BOTTOM_RIGHT, -70, 0);
}

static lv_res_t keypad_btn_press(lv_obj_t * btn)
{
	last_key = lv_obj_get_free_num(btn);    /*Save the key*/
	last_state = LV_INDEV_STATE_PR;         /*Save the state*/

	return LV_RES_OK;
}

/**
 * Called when the Next button is released
 * @param btn pointer to the button
 * @param indev_proc pointer to the caller display input
 * @return LV_RES_OK: because the button is not deleted
 */
static lv_res_t keypad_btn_release(lv_obj_t * btn)
{
	(void)btn;      /*Unused*/

	last_state = LV_INDEV_STATE_REL;        /*Save the new state but leave the last key*/

	return LV_RES_OK;
}

/**
* Initialize the Hardware Abstraction Layer (HAL) for the Littlev graphics library
*/
static void hal_init(void)
{
	/* Add a display
	* Use the 'monitor' driver which creates window on PC's monitor to simulate a display*/
	monitor_init();
	lv_disp_drv_t disp_drv;
	lv_disp_drv_init(&disp_drv);            /*Basic initialization*/
	disp_drv.disp_flush = monitor_flush;
	disp_drv.disp_fill = monitor_fill;
	disp_drv.disp_map = monitor_map;
	lv_disp_drv_register(&disp_drv);

	/* Add the mouse (or touchpad) as input device
	* Use the 'mouse' driver which reads the PC's mouse*/
	mouse_init();
	lv_indev_drv_t indev_drv;
	lv_indev_drv_init(&indev_drv);          /*Basic initialization*/
	indev_drv.type = LV_INDEV_TYPE_POINTER;
	indev_drv.read = mouse_read;         /*This function will be called periodically (by the library) to get the mouse position and state*/
	lv_indev_drv_register(&indev_drv);

	/* If the PC keyboard driver is enabled in`lv_drv_conf.h`
	* add this as an input device. It might be used in some examples. */
#if USE_KEYBOARD
	lv_indev_drv_t kb_drv;
	kb_drv.type = LV_INDEV_TYPE_KEYPAD;
	kb_drv.read = keyboard_read;
	kb_indev = lv_indev_drv_register(&kb_drv);
#endif

	/* Tick init.
	* You have to call 'lv_tick_inc()' in every milliseconds
	* Create an SDL thread to do this*/
	SDL_CreateThread(tick_thread, "tick", NULL);
}

/**
* A task to measure the elapsed time for LittlevGL
* @param data unused
* @return never return
*/
static int tick_thread(void *data)
{
	while (1) {
		lv_tick_inc(1);
		SDL_Delay(1);   /*Sleep for 1 millisecond*/
	}

	return 0;
}
```

**Comment by embeddedt** [comment]
A note from debugging: It seems that everything always (not just one time) works as expected with buttons 1 and 2.

**Comment by MichaelVLV** [comment]
Any ideas what going with buttons? Could it be that "switcher" chooses labels instead of button objects?

**Comment by embeddedt** [comment]
@MichaelVLV No. I'll ask @kisvegabor to comment.

**Comment by kisvegabor** [comment]
Hi,

I found the issue. If you copy an object which is added to a group, the new (copied) object will be added to the same group automatically ([code](https://github.com/littlevgl/lvgl/blob/master/lv_core/lv_obj.c#L308)).

I'm not sure which way is more intuitive. (automatic adding or not). What do you think?

**Comment by embeddedt** [comment]
@kisvegabor I think the automatic adding makes more sense, since one expects an exact copy of the object. What I think should be done is blocking an object from being added to the same group twice. That looks like the issue in this case.

**Comment by kisvegabor** [comment]
@embeddedt I agree!

**Comment by kisvegabor** [comment]
I'll fix it soon in `v6.0`

**Comment by kisvegabor** [comment]
Fixed in `dev-6.0`. If still not working feel free to reopen this issue.
